File: SIMPLEAPI_Demo/SIMPLEAPI_Demo/ItemBoleta.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIMPLEAPI_Demo
{
    public class ItemBoleta
    {
        public string Nombre { get; set; }
        public decimal Cantidad { get; set; }
        public int Precio { get; set; }
        public int Total { get { return (int)Math.Round(Cantidad * Precio, 0); } }
        public bool Afecto { get; set; }
        public string UnidadMedida { get; set; }
        public double Descuento { get; set; }
        public int _tipoImpuesto;
        public SimpleAPI.Enum.TipoImpuesto.TipoImpuestoEnum TipoImpuesto
        {
            get { return _tipoImpuesto != 0 ? (SimpleAPI.Enum.TipoImpuesto.TipoImpuestoEnum)_tipoImpuesto : SimpleAPI.Enum.TipoImpuesto.TipoImpuestoEnum.NotSet; }
            set { _tipoImpuesto = (int)value; }
        }
    }
}

File: SIMPLEAPI_Demo/SIMPLEAPI_Demo_VB/ItemBoleta.vb
Public Class ItemBoleta
    Public Property Nombre As String
    Public Property Cantidad As Decimal
    Public Property Precio As Integer

    Public ReadOnly Property Total As Integer
        Get
            Return CInt(Math.Round(Cantidad * Precio, 0))
        End Get
    End Property

    Public Property Afecto As Boolean
    Public Property UnidadMedida As String
    Public Property Descuento As Double
    Public _tipoImpuesto As Integer

    Public Property TipoImpuesto As ChileSystems.DTE.Engine.[Enum].TipoImpuesto.TipoImpuestoEnum
        Get
            Return If(_tipoImpuesto <> 0, CType(_tipoImpuesto, ChileSystems.DTE.Engine.[Enum].TipoImpuesto.TipoImpuestoEnum), ChileSystems.DTE.Engine.[Enum].TipoImpuesto.TipoImpuestoEnum.NotSet)
        End Get
        Set(ByVal value As ChileSystems.DTE.Engine.[Enum].TipoImpuesto.TipoImpuestoEnum)
            _tipoImpuesto = CInt(value)
        End Set
    End Property
End Class

